src/Core/src/Platform/Windows/MauiButton.cs

```diff
@@ -2,19 +2,36 @@
 using Microsoft.UI.Xaml.Automation.Peers;
 using Microsoft.UI.Xaml.Controls;
 using Microsoft.UI.Xaml.Media;
-using Windows.Security.Credentials.UI;
 using WThickness = Microsoft.UI.Xaml.Thickness;
 
 namespace Microsoft.Maui.Platform
 {
 	public class MauiButton : Button
 	{
+		DefaultMauiButtonContent _defaultButtonContent;
+
 		public MauiButton()
 		{
-			Content = new DefaultMauiButtonContent();
+			Content = _defaultButtonContent = new DefaultMauiButtonContent();
 
 			VerticalAlignment = VerticalAlignment.Stretch;
 			HorizontalAlignment = HorizontalAlignment.Stretch;
+
+			SizeChanged += MauiButton_SizeChanged;
+		}
+
+		private void MauiButton_SizeChanged(object sender, SizeChangedEventArgs e)
+		{
+			if (!double.IsInfinity(Width))
+			{
+				_defaultButtonContent.LimitImageWidth(
```

Comment:
If the content is replaced, this will still be updating and also never collected. Is there a way in which we can do Content is DefaultButtonContent content? Then we only update if it still our content.

Reply:
Yeah I can switch to that. I didn't do it at first because I was trying to avoid that cast on every resize